app/services/ems_physical_infra_dashboard_service.rb

```diff
@@ -195,8 +230,9 @@ def check_num(val)
 
   def recent_records(model)
     all_records = Hash.new(0)
-    records = model.where('created_at > ? and ems_id = ?', 30.days.ago.utc, @ems.id)
-    records = records.includes(:resource => [:ext_management_system]) if @ems.blank?
+    records = model.where('created_at > ?', 30.days.ago.utc)
+    records.where('ems_id = ?', @ems_id) if @ems_id.present?
```

Comment:
looks like an `if/else` would be more readable here as these are really mutually exclusive